src/UnigramWordPredictor.java

```diff
@@ -49,9 +50,25 @@ public UnigramWordPredictor(Tokenizer tokenizer) {
    * @param scanner the Scanner to read the training text from
    */
   public void train(Scanner scanner) {
+    neighborMap = new HashMap<String, List<String>>(); // instantiate neighborMap as HashMap
     List<String> trainingWords = tokenizer.tokenize(scanner);
 
-    // TODO: Convert the trainingWords into neighborMap here
+    for (int i = 0; i < trainingWords.size(); i++) {
+      String current = trainingWords.get(i); // save current token in arraylist
+      String next = null;
+
+      if (i < trainingWords.size()-1) {
+        next = trainingWords.get(i+1); // save follow-up token in arraylist
+      }
```

Comment:
Nice check! As an alternative, can you consider how you could modify your for loop so this check is not necessary?